python_ta/contracts/__init__.py

```diff
@@ -183,6 +184,24 @@ def _check_function_contracts(wrapped, instance, args, kwargs):
                 f"return type annotation {_display_annotation(return_type)}"
             )
 
+    # Check function postconditions
+    pre_replacement_postconditions = parse_assertions(wrapped, parse_token="Postcondition")
+    legal_return_val_var_name = _get_legal_return_val_var_name(
+        {**wrapped.__globals__, **function_locals}
+    )
+    postconditions = [
+        assertion.replace(FUNCTION_RETURN_VALUE, legal_return_val_var_name)
+        for assertion in pre_replacement_postconditions
+    ]
+
+    _check_assertions(
+        wrapped,
+        {**function_locals, legal_return_val_var_name: r},
```

Comment:
The overall approach is great, but in terms of the implementation:

1. I think it's better to do the replacing inside `_check_assertions`. So rather than changing the `function_locals` and passing the identifier name, keep the original locals and pass the return value.
2. This way, in the error message, you can separate the arguments and the return value---this is important because the return value is not technically an "argument", and we want to make sure that PythonTA respects proper terminology. I could see an error message like `with arguments {...} and return value ...`
3. As a bonus, you'll have access to the original assertion in `_check_assertions`, so there won't be a need to do the reversing replacement.